File: mobile/test/core/utils/display_name_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:ora/core/utils/display_name.dart';

void main() {
  test('trims and collapses whitespace', () {
    expect(DisplayName.normalize('  Ada   Khan  '), 'Ada Khan');
  });

  test('rejects empty and whitespace-only names', () {
    expect(DisplayName.validationError(''), isNotNull);
    expect(DisplayName.validationError('   '), isNotNull);
  });

  test('rejects too-short and too-long names', () {
    expect(DisplayName.validationError('A'), isNotNull);
    expect(DisplayName.validationError('A' * 51), isNotNull);
  });

  test('rejects control characters and digit-only input', () {
    expect(DisplayName.validationError('Ada\u0007Khan'), isNotNull);
    expect(DisplayName.validationError('12'), isNotNull);
  });

  test('accepts a valid two-letter name', () {
    expect(DisplayName.validationError('Li'), isNull);
    expect(DisplayName.normalize('Li'), 'Li');
  });
}

File: backend/auth-service/src/__tests__/display_name.test.ts
import { describe, expect, it } from 'vitest';
import { normalizeDisplayName } from '../services/display_name';

describe('normalizeDisplayName', () => {
  it('trims, NFC-normalizes, and collapses whitespace', () => {
    expect(normalizeDisplayName('  Ada   Khan  ')).toBe('Ada Khan');
  });

  it('rejects empty and whitespace-only values', () => {
    expect(() => normalizeDisplayName('')).toThrow(/required/i);
    expect(() => normalizeDisplayName('   ')).toThrow(/required/i);
  });

  it('rejects non-strings', () => {
    expect(() => normalizeDisplayName(1)).toThrow(/string/i);
  });

  it('rejects too-short, too-long, control, and non-letter names', () => {
    expect(() => normalizeDisplayName('A')).toThrow(/at least 2/i);
    expect(() => normalizeDisplayName('A'.repeat(51))).toThrow(/at most 50/i);
    expect(() => normalizeDisplayName('Ada\u0007Khan')).toThrow(/invalid/i);
    expect(() => normalizeDisplayName('12')).toThrow(/two letters/i);
  });
});
